comms/email/mockemail/mockemail.go

```diff
@@ -16,11 +17,12 @@ func NewClient() *Client {
 	}
 }
 
-func (c *Client) Send(to []string, message json.RawMessage, tplID email.TplID) error {
+func (c *Client) Send(to []string, cc []string, message json.RawMessage, tplID email.TplID) error {
 
 	for _, v := range to {
 		c.sendLogs = append(c.sendLogs, SendLog{
 			to:      v,
+			cc:      strings.Join(cc, ","),
```

Comment:
This was actually a really clever way to meet the requirements within the codebase as it currently stands. But I'd consider this an anti-pattern, because we are overloading what looks like a single value field with multiples, and that makes for a confusing data contract because now callers will be expected to know about this field overload and attempt to deserialize the string (which the test you wrote that uses this is not doing btw). Also a change in the join algorithm could then break anything that's using this.

Given that refactoring any internal parts of the codebase is always fair game (changing public data/api contracts is not), what would you suggest could be a better way(s) to handle the what you're encountering in the SendLog?

Reply:
Hey Greg! Thanks for the review and I completely agree the join is absolutely a weak point. There are two better approaches that I like for alternative reasons:

1. Matching `sendgrids` implementation of a single from email with all constituents (TOs, CCs, BCCs) attached. The `SendLog` would then contain a list of `tos` and a list of `ccs`.

Example:
```go
// Sendlog
type SendLog struct {
	to      []string
	cc      []string
	tplID   email.TplID
	message json.RawMessage
}
...
// Mockmail Send
c.sendLogs = append(c.sendLogs, SendLog{
			to:      to,
			cc:      cc,
			message: message,
			tplID:   tplID,
})
...
// Testcase
if !slices.Equal(lastEmail.ExtractTos(), tc.payload.EmailTo) {
    t.Fatalf("expected %v but got %v", lastEmail.ExtractTos(), tc.payload.EmailTo) 
}

if !slices.Equal(lastEmail.ExtractCcs(), tc.payload.EmailCc) {
    t.Fatalf("expected %v but got %v", lastEmail.ExtractCcs(), tc.payload.EmailCc) 
}

```

Something like this would provide a condensed `SendLog` and allow a good range of testability while following `sendgrids` implementation.

2. The alternative would be a verbose ( possibly extreme in this scenario ) `SendLog` which follows email semantics of To,CC,BCC being completely separate. This veers from alignment with the `sendgrid` implementation, however if the `SendLog` were to be more useful e.a telemetry, or data export, and analysis having a unique log for each constituent would be beneficial.

Example:
```go
// Sendlog
type SendLog struct {
	to           string
        isCc         bool
	tplID        email.TplID
	message      json.RawMessage
}
```

I don't want to waste too much time with this example as I noticed in the following comment you mentioned `SendLog` is _only_ here for tests and to replicate sendgrid functionality. That being the case I'll just leave this here as brain food.

Please let me know what you think of the design approach👍

Reply:
@DLzer Approach 1 is exactly what I was looking for. The reasoning being that the purpose of the mockemail package is simply to allow testing of the sendgrid implementation, so it makes sense that the log structure would match the one-log-per-email pattern that the sendgrid implementation uses rather than a log entry per recipient type of log which we don't really care about here.

Reply:
@gregtzar-glovebox Awesome, thanks for the update! I've made a push with the changes and confirmed tests pass business requirements 👍